### ISSUE TITLE
Assertion `!fmt->p_free' failed.

### ISSUE THREAD

**Issue by mmoisse** [Illumina/agg#16]
When I try to extract a specific region from chunks it usually works, but for this specific region it failed.
Could you help me out?

```
$ /usr/bin/time agg genotype -r chr1:1558780-1558785 chunks/chunk_0?.bcf \
     -Ov -o chunks/tmp/merged.chr1-1558768-1559079.bcf --thread 1
chunks/chunk_01.bcf chunks/chunk_02.bcf 
n_threads = 1
1558780
interval_start = 1558779 interval_end=1558784
543 samples
Writing output to chunks/tmp/merged.chr1-1558768-1559079.BE_543.bcf
agg: vcf.c:2988: bcf_update_format: Assertion `!fmt->p_free' failed.
Command terminated by signal 6
0.14user 0.01system 0:00.23elapsed 69%CPU (0avgtext+0avgdata 57876maxresident)k
0inputs+24outputs (0major+13900minor)pagefaults 0swaps
```

The odd thing is that when I perform the same command but on only one chunk it works without any problems. The first chunk returns 1 variant, while the second chunk contains no variants:

```
chr1    1558782 .       C       T       70      .       AN=842;AC=1;PF=1;AD=9516,7;DP=9523
```

I belief it has something to do with indels because I just noticed that if I enlarge the region a bit to the left, I don't get the error

```
$ /usr/bin/time agg genotype -r chr1:1558754-1558785 chunks/chunk_0?.bcf \
     -Ov -o chunks/tmp/merged.chr1-1558768-1559079.bcf --thread 1
chunks/chunk_01.bcf chunks/chunk_02.bcf 
n_threads = 7
1558754
interval_start = 1558753 interval_end=1558784
543 samples
Writing output to chunks/tmp/merged.chr1-1558768-1559079.BE_543.bcf
0.12user 0.02system 0:00.14elapsed 95%CPU (0avgtext+0avgdata 65236maxresident)k
```

And the vcf looks like this

```
chr1    1558754 .       ACCCTCTGCCCACAGGTCCCGAGCAGCCCGGCCCACCATGGAC     A       391     .       AN=1086;AC=1;PF=1;AD=9741,10;DP=9751
chr1    1558782 .       C       T       70      .       AN=1086;AC=1;PF=1;AD=13284,7;DP=13291
```
## Version info:

I'm using agg version v0.3.0-develop-2-g942d857
for the genotype command I also tried v0.3.3.dev-31-gaa44755 but it didn't solve the issue.

**Comment by jaredo** [comment]
I think I see the problem, the indel spans the range `chr1:1558754-1558797` but you have requested a smaller range `chr1:1558754-1558785`. I suspect what then happens is that agg has not queried the full length of the deletion from its `.dpt` file meaning it is probably accessing some invalid memory, this finally yields an assert fail in htslib.

This is a bug and I will fix it, but probably will not have time in the next few weeks.

In the interim, I would suggest specifying entire chromosomes as regions, this will completely circumvent any problems with variants exceeding the specified range (not the first time I have seen a bug like this). 

I would also encourage you to use the latest version, the software has had a lot of bug fixes!

best,

Jared

**Comment by mmoisse** [comment]
Thank you, I also believe the bug has something to with spanning indels, more specifically when the start site of a spanning indel is not in the requested region:
chr1:1558780-1558785  >  crahes
chr1:1558754-1558785  >  returns indel chr1:1558754-1558797

**Comment by jaredo** [comment]
Yes, basically I need to add some buffer on either side of the specified region to protect against these problems.  I usually just run whole chromosomes at a time so had not considered this use case.

j

**Comment by mmoisse** [comment]
Running a whole chromosome takes some time, therefore I only wanted to extract the regions of interest.

I made a quick fix in which I first look if there are indels in the region I want and enlarge it if needed:

    chr=chr1
    begin=1558780
    end=1558785
    chunks="../chunks/chunk_0?.bcf"
    
    begin=`for bcf in ${chunks}; 
    do
       bcftools view -H -r $chr:${begin}-${end} $bcf | cut -f 2 | head -n 1
    done | sort -n | head -n 1`
    
    /usr/bin/time agg genotype -r $chr:${begin}-${end} ${chunks} \
     -Ov -o chunks/tmp/merged.$chr-${begin}-${end}.bcf --thread 1

**Comment by jaredo** [comment]
Sorry for the delay. This should now be resolved in #20 , please re-open if you can still recreate the problem.

Agg will now only return variants that are not *fully* contained within a specified interval.
